app/src/main/java/ru/vssemikoz/newsfeed/data/RemoteNewsRepository.java

```diff
@@ -1,49 +1,40 @@
 package ru.vssemikoz.newsfeed.data;
 
-import org.jetbrains.annotations.NotNull;
+import java.util.List;
 
 import javax.inject.Inject;
 
-import retrofit2.Call;
-import retrofit2.Callback;
-import retrofit2.Response;
+import io.reactivex.rxjava3.core.Single;
 import ru.vssemikoz.newsfeed.AppConfig;
 import ru.vssemikoz.newsfeed.api.NewsApi;
+import ru.vssemikoz.newsfeed.data.mappers.NewsMapper;
 import ru.vssemikoz.newsfeed.models.Category;
-import ru.vssemikoz.newsfeed.models.NewsApiResponse;
+import ru.vssemikoz.newsfeed.models.NewsFeedParams;
+import ru.vssemikoz.newsfeed.models.NewsItem;
 
 public class RemoteNewsRepository implements NewsRepository {
     @Inject
     NewsApi newsApi;
     @Inject
     AppConfig config;
+    @Inject
+    NewsMapper mapper;
 
     @Inject
     public RemoteNewsRepository() {
     }
 
     @Override
-    public void getNewsFiltered(Category category, RequestListener listener) {
+    public Single<List<NewsItem>> getNewsFiltered(NewsFeedParams params) {
+        Category category = params.getFilter().getCategory();
         String categoryKey = null;
-        Call<NewsApiResponse> call;
         if (category != Category.ALL) {
             categoryKey = Category.getCategoryName(category);
         }
-        call = newsApi.getNews(config.getCountryKey(), categoryKey, config.getApiKey());
-        call.enqueue(getNewsApiCallBack(listener));
-    }
-
-    private Callback<NewsApiResponse> getNewsApiCallBack(NewsRepository.RequestListener listener) {
-        return new Callback<NewsApiResponse>() {
-            @Override
-            public void onResponse(@NotNull Call<NewsApiResponse> call, @NotNull Response<NewsApiResponse> response) {
-                listener.onRequestSuccess(response);
-            }
 
-            @Override
-            public void onFailure(@NotNull Call<NewsApiResponse> call, @NotNull Throwable t) {
-                listener.onRequestFailure(t);
-            }
-        };
+        return Single.just(newsApi.getNews(config.getCountryKey(), categoryKey, config.getApiKey()))
```

Comment:
You're wrapping a Single, returned from the `newsApi.getNews` to a new `Single`. You should directly call `newsApi.getNews` here without wrapping it up into the `Single.just`. Then you'd not need a `.flatMap` operator.